spt/features/visualizations/player_trace/import_export/tr_binary.hpp

```diff
@@ -4,54 +4,159 @@
 
 #ifdef SPT_PLAYER_TRACE_ENABLED
 
-extern const char* SPT_VERSION;
+#include "spt/utils/game_detection.hpp"
 
 namespace player_trace
 {
-	class ITrWriter
+	constexpr char TR_FILE_ID[8] = "sausage";
+	constexpr uint32_t TR_SERIALIZE_VERSION = 2;
+	constexpr size_t TR_MAX_SPT_VERSION_LEN = 32;
+
+	struct TrPreamble
+	{
+		char fileId[sizeof TR_FILE_ID];
+		uint32_t fileVersion;
+	};
+
+	struct TrHeader_v1
+	{
+		char lastExportSptVersion[TR_MAX_SPT_VERSION_LEN];
+		uint32_t lumpsOff;
+		uint32_t nLumps;
+		uint32_t numRecordedTicks;
+		TrIdx<TrAbsBox> playerStandBboxIdx, playerDuckBboxIdx;
+	};
+
+	struct TrHeader_v2 : TrHeader_v1
 	{
-	public:
-		template<typename T>
-		bool Write(const T& o)
+		TrHeader_v2() {}
+		TrHeader_v2(const TrHeader_v1& v1)
 		{
-			return Write(std::span{(const std::byte*)&o, sizeof o});
+			memcpy(this, &v1, sizeof v1);
+
+			strncpy(gameInfo.gameName, "<NOT_RECORDED>", sizeof gameInfo.gameName);
+			strncpy(gameInfo.modName, "<NOT_RECORDED>", sizeof gameInfo.modName);
+			gameInfo.gameVersion = -666;
+			strncpy(playerName, "<NOT_RECORDED>", sizeof playerName);
 		}
 
-		virtual bool Write(std::span<const std::byte> sp) = 0;
-		virtual bool DoneWritingTrace() = 0; // notify writer to flush
+		// a bunch of info from when the trace was first recorded
+
+		struct
+		{
+			char gameName[16]; // max length returned by utils::GetGameName()
+			char modName[260]; // MAX_OSPATH
+			int32_t gameVersion;
+		} gameInfo;
+
+		char playerName[32]; // MAX_PLAYER_NAME_LENGTH
```

Comment:
I think these char arrays are missing one byte to account for the null terminator. strncpy does not null-terminate the string if it runs out of space. Very comical design decision on their part, got em good.